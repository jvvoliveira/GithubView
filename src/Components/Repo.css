.repo{
    display: block;
    margin: 30px;
    padding: 10px;
    border: solid;
}