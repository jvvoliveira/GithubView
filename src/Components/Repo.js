import React from "react";
import styles from "./Repo.module.scss";

const Repo = props => {
  return (
    <a className={styles.repositorio} href={props.link} target="_blank">
      <h3 data-testid='name'>{props.name}</h3>
      <div className={styles.text}>
        <p data-testid='description'>{props.description}</p>
      </div>
      <p>{props.language}</p>
    </a>
  );
};

export default Repo;
