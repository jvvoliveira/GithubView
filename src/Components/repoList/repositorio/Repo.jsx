import React from "react";
import styles from "./Repo.module.scss";

const Repo = props => {
  const { name, language, description, link, index } = props;
  return (
    <a
      data-testid={`repo-${index}`}
      className={styles.repositorio}
      href={link}
      target="_blank"
      key={index}
    >
      <h3 data-testid="name">{name}</h3>
      <div className={styles.text}>
        <p data-testid="description">{description}</p>
      </div>
      <p>{language}</p>
    </a>
  );
};

export default Repo;
