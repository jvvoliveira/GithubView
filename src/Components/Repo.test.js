import Repo from './Repo';
import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import renderer from 'react-test-renderer';

describe("test initial Repo", ()=>{
    it("should renders without crashing", ()=>{
        const div = document.createElement('div');
        ReactDOM.render(<Repo />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it("snapshot",()=>{
        const tree = renderer.create(<Repo></Repo>).toJSON();
        expect(tree).toMatchSnapshot();
        
    });
})