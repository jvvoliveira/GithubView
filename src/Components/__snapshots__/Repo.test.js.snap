// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test initial Repo snapshot 1`] = `
<div
  className="repo"
>
  <h3 />
  <p />
  <a
    target="_blank"
  >
    reposit√≥rio
  </a>
  <p>
    Linguagem: 
  </p>
  <br />
</div>
`;
